fish_completions.go

```diff
@@ -141,19 +142,24 @@ function __%[1]s_prepare_completions
 
     set -l nospace (math (math --scale 0 $directive / $shellCompDirectiveNoSpace) %% 2)
     set -l nofiles (math (math --scale 0 $directive / $shellCompDirectiveNoFileComp) %% 2)
+    set -l nomatching (math (math --scale 0 $directive / $shellCompDirectiveNoMatching) %% 2)
 
-    __%[1]s_debug "nospace: $nospace, nofiles: $nofiles"
+    __%[1]s_debug "nospace: $nospace, nofiles: $nofiles, nomatching: $nomatching"
 
     # If we want to prevent a space, or if file completion is NOT disabled,
     # we need to count the number of valid completions.
     # To do so, we will filter on prefix as the completions we have received
     # may not already be filtered so as to allow fish to match on different
     # criteria than the prefix.
     if test $nospace -ne 0; or test $nofiles -eq 0
```

Comment:
This is a special case which the comment may not have made very clear.

Let's take file completion.  We only want to trigger file completion if there are no other completions available (from the point of view of the user).  This is not normally what fish does, but it is how the other shells behave so we decided to standardize on that.

However, if the program does not filter on prefix (to allow fuzzy matching), it will always return some completions and we will not be able to know if fish will match any of them or not; so, we would not know if there are no real completions and couldn't tell if we should trigger file completion.  Therefore, in this case, we turn off fuzzy matching and explicitly match on prefix; if there are no completions left, we know to turn on file completion.

A similar scenario applies to the `nospace` directive.  You can refer to #1249 for details.

This means that in this case, we cannot do fuzzy completion, or it would break the request from the program to do file completion and/or nospace completion.

I don't know if that was clear, so don't hesitate to ask for clarification.

Reply:
Hum I see, what do you suggest changing the logic to in that case ?